pcd8544/pcd8544.go

```diff
@@ -0,0 +1,119 @@
+// Package pcd8544 implements a driver for the PCD8544 led matrix controller, used in Nokia's 5110 and 3310 phones.
+//
+// Datasheet: http://eia.udg.edu/~forest/PCD8544_1.pdf
+package pcd8544
+
+import (
+	"time"
+
+	"github.com/aykevl/tinygo/src/machine"
+)
+
+// Device wraps an SPI connection.
+type Device struct {
+	bus    machine.SPI
+	dcPin  machine.GPIO
+	rstPin machine.GPIO
+	scePin machine.GPIO
+	buffer [504]byte
+}
+
+// New creates a new PCD8544 connection. The SPI bus must already be configured.
+func New(bus machine.SPI, dcPin machine.GPIO, rstPin machine.GPIO, scePin machine.GPIO) *Device {
+	return &Device{
+		bus:    bus,
+		dcPin:  dcPin,
+		rstPin: rstPin,
+		scePin: scePin,
+		buffer: [504]byte{},
+	}
+}
+
+// Init initializes the LCD with default configuration
+func (d *Device) Init() {
+	d.rstPin.Low()
+	time.Sleep(500 * time.Millisecond)
```

Comment:
I don't think this is necessary on most chips, and 500ms is certainly way more than required. The required reset time is at least 100ns. As peripheral writes are generally 2 cycles on Cortex-M, this means that not sleeping at all between .Low() and .High() translates into at least 125ns for a 16MHz chip (possibly more depending on the compiler and optimization level).
To future proof this library, I would recommend simply sleeping for 100ns literally, so `time.Sleep(100 * time.Nanosecond)`. Or maybe a little bit more if you want to be sure. That avoids half a second of startup time.

Reply:
I did not appreciate any difference between using more or less time (or any at all), it looked too much for me too but Adafruit's pcd8544 library has a 500ms delay. Will change this.